### PR TITLE
Honor test fixture compiler flags in browser tests. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Also fix all warnings resuling in `-Werror` being added.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -8257,11 +8257,15 @@ def setUp(self):
...
+      '-Wno-dynamic-class-memaccess',
+      '-Wno-format',
+      '-Wno-format-extra-args',
+      '-Wno-format-security',
+      '-Wno-pointer-bool-conversion',
+      '-Wno-unused-volatile-lvalue',
       '-Wno-c++11-compat-deprecated-writable-strings',
-      '-Wno-invalid-pp-token', '-Wno-shift-negative-value'
+      '-Wno-invalid-pp-token',
+      '-Wno-shift-negative-value'
```
Comment (sbc100) [inline tests/test_core.py:8268]: This is just a re-format.. no flags changing.

File: tests/test_browser.py [hunk 2]
```diff
@@ -155,6 +155,14 @@ def setUpClass(cls):
     print('Running the browser tests. Make sure the browser allows popups from localhost.')
     print()
 
+  def setUp(self):
+    super(BrowserCore, self).setUp()
+    # avoid various compiler warnings in our tests contain
```
Comment (kripken) [inline tests/test_browser.py:160]:
```suggestion
    # avoid various compiler warnings in our tests
```

File: tests/test_browser.py [hunk 3]
```diff
@@ -1725,7 +1733,7 @@ def test_glgears_deriv(self):
   @requires_graphics_hardware
   def test_glbook(self):
     self.emcc_args.remove('-Werror')
-    self.emcc_args += ['-Wno-pointer-sign', '-Wno-int-conversion']
+    self.emcc_args += ['-Wno-pointer-sign']
```
Comment (kripken) [inline tests/test_browser.py:1736]:
```suggestion
```

Comment (kripken) [reply tests/test_browser.py:1736]: that is set globally now?

**Review by kripken — APPROVED** [review]

**Pull request merged**